e2e/actions/actions.go

```diff
@@ -755,155 +716,273 @@ func (c actionTests) PersistentOverlay(t *testing.T) {
-			exit:    0,
-			profile: e2e.RootProfile,
+			name:     "overlay_multiple_find_ext3",
+			argv:     []string{"--overlay", "${EXT3_DIR}", "--overlay", "${SQUASH_DIR}" + ":ro", c.env.ImagePath, "test", "-f", "/multiple_overlay_fs"},
+			fakeroot: true,
+			exit:     0,
 		},
 		{
-			name:    "overlay_multiple_find_squashfs",
-			argv:    []string{"--overlay", ext3Img, "--overlay", squashfsImage + ":ro", c.env.ImagePath, "test", "-f", fmt.Sprintf("/%s", squashMarkerFile)},
-			exit:    0,
-			profile: e2e.RootProfile,
+			name:     "overlay_multiple_find_squashfs",
+			argv:     []string{"--overlay", "${EXT3_DIR}", "--overlay", "${SQUASH_DIR}" + ":ro", c.env.ImagePath, "test", "-f", "/squash_marker"},
+			fakeroot: true,
+			exit:     0,
 		},
 		{
-			name:    "overlay_noroot",
-			argv:    []string{"--overlay", dir, c.env.ImagePath, "true"},
-			exit:    255,
-			profile: e2e.UserProfile,
+			name: "overlay_noroot",
+			argv: []string{"--overlay", "${OVERLAY_DIR}", c.env.ImagePath, "true"},
+			exit: 0,
 		},
 		{
-			name:    "overlay_noflag",
-			argv:    []string{c.env.ImagePath, "test", "-f", "/foo_overlay"},
-			exit:    1,
-			profile: e2e.RootProfile,
+			name: "overlay_noflag",
+			argv: []string{c.env.ImagePath, "test", "-f", "/foo_overlay"},
+			exit: 1,
 		},
 		{
 			// https://github.com/apptainer/singularity/issues/4329
-			name:    "SIF_writable_without_overlay_partition_issue_4329",
-			argv:    []string{"--writable", c.env.ImagePath, "true"},
-			exit:    255,
-			profile: e2e.RootProfile,
+			name: "SIF_writable_without_overlay_partition_issue_4329",
+			argv: []string{"--writable", c.env.ImagePath, "true"},
+			exit: 255,
 		},
 		{
 			// https://github.com/apptainer/singularity/issues/4270
-			name:    "overlay_dir_relative_path_issue_4270",
-			argv:    []string{"--overlay", filepath.Base(dir), sandboxImage, "test", "-f", "/dir_overlay"},
-			dir:     filepath.Dir(dir),
-			exit:    0,
-			profile: e2e.RootProfile,
+			name:     "overlay_dir_relative_path_issue_4270",
+			argv:     []string{"--overlay", "${BASE_OVERLAY_DIR}", "${SANDBOX_DIR}", "test", "-f", "/dir_overlay"},
+			dir:      "${PARENT_OVERLAY_DIR}",
+			fakeroot: true,
+			exit:     0,
+		},
+		{
+			name: "centos 7 Embedded overlay partition in SIF",
+			argv: []string{"${CENTOS7_DIR}", "ps"},
+			exit: 0,
 		},
 	}
 
-	for _, tt := range tests {
-		c.env.RunApptainer(
-			t,
-			e2e.AsSubtest(tt.name),
-			e2e.WithProfile(tt.profile),
-			e2e.WithDir(tt.dir),
-			e2e.WithCommand("exec"),
-			e2e.WithArgs(tt.argv...),
-			e2e.ExpectExit(tt.exit),
-		)
+	// centos7 img will only create one time
+	centos7Dir := c.centosCreate(t, testdir)
+	for _, profile := range profiles {
+		// create an overlay directory
+		overlayDir := c.overlayDirCreate(t, testdir)
+		squashDir := c.squashCreate(t, testdir)
+		ext3Dir := c.ext3Create(t, testdir)
+		sandboxDir := c.sandboxCreate(t, testdir)
```

Comment:
Why do these need to be recreated for every profile?

Reply:
originally, this is because some of these test cases will create overlays using different profiles, overlays created by the root will not be accessed by others. I planned to clean them all after each loop (profile loop). Anyway, we do not need this if the profiles are ordered. So I've refactored the code and these dirs won't need to be created any more.